### DISCUSSION TITLE
Issue with pandas `reset_index()`

### DISCUSSION THREAD

**Discussion by JanEricNitschke** [microsoft/pyright#8209] [Q&A] [answered]
When doing this

```
import pandas as pd

pd.DataFrame().reset_index(name="kills")
```

I get the following error from pyright

```
error: No overloads for "reset_index" match the provided arguments
    Argument types: (Literal['kills']) (reportCallIssue)
```

I am not sure if this is an issue with pyright (1.1.368), pandas-stubs (2.2.2.240603) or on my side.

**Comment by erictraut** [comment]
According to the stubs, there is no keyword parameter called `name`. There is one called `names`. Perhaps that's what you intended?

**Reply by JanEricNitschke** [reply]
Ah, i think i made a mistake when trying to get this to a minimal example. The original code is this:

```
    kills_ct = (
        demo.kills[demo.kills["attacker_team_name"] == "CT"]
        .groupby(["attacker_name", "attacker_steamid"])
        .size()
        .reset_index(name="kills")
        .rename(columns={"attacker_name": "name", "attacker_steamid": "steamid"})
    )
```

And there the referenced `reset_index()` is the one for [pd.Series](https://github.com/pandas-dev/pandas-stubs/blob/main/pandas-stubs/core/series.pyi#L443).

Regardless, the exact same message comes from pyright if i change it to names.

**Reply by erictraut** [reply]
Can you provide a corrected, minimal, self-contained example?

**Reply by JanEricNitschke** [reply]
```
import pandas as pd

kills = pd.DataFrame({"attacker_name": ["1", "2","3"]})
kills_total = (
    kills.groupby(["attacker_name"])
    .size()
    .reset_index(name="kills")
)
```

This runs normally but gives the original error.

**Reply by erictraut** [reply, marked as answer]
The type stub indicates that the `size` method returns a value of type `DataFrame | Series[int]`. The `reset_index` method is valid for a `Series[int]` object, but it's not for a `DataFrame` object because of the parameter name mismatch that I mentioned above.

So this isn't a bug in pyright. You may want to ask the maintainers of pandas-stubs for additional help, as I don't know much about the pandas library.

**Reply by JanEricNitschke** [reply]
Ah, ok. Had a look and it seems that the return type depends on internals of the GroupBy result that can not be easily handled using the type system.

Will just ignore the errors for now.

Thanks for the help!
